### ISSUE TITLE
Deprecating CompositeObservable in favor of Subjects

### ISSUE THREAD

**Issue by thomasnield** [ReactiveX/RxJavaFX#54]
The [`CompositeObservable`](https://thomasnield.gitbooks.io/rxjavafx-guide/content/10.%20Decoupling.html) was somewhat of a moving target after I first had the idea. It was a tool to safely decouple RxJava streams in a JavaFX application without exposing a `Subject`, which can [encourage antipatterns especially with Rx newbies](http://tomstechnicalblog.blogspot.com/2016/03/rxjava-problem-with-subjects.html). 

Although I haven't changed my sentiments on Subjects, I may have softened my stance a little when it comes to decoupling. The `CompositeObservable` ultimately came to resemble a `Subject` without exposing the `Observer` methods. Not to mention, it is somewhat inconvenient having to call its `toObservable()` method to get the `Observable`. 

Therefore, I am considering deprecating the `CompositeObservable` and telling people to use `PublishSubject`, `ReplaySubject`, etc instead. If anybody has thoughts on this topic, please feel free to share.

**Comment by rguillens** [comment]
I've tested different scenarios, and I can say `Subject `is a most straight forward solution than `CompositeObservable`. I'm just re-thinking about all common scenarios involving concurrency , although ideally `CompositeObservable `should be used to listen UI events from other layer, as events are always produced on JavaFx Application Thread, theorically there should be no problem.

**Comment by thomasnield** [comment]
Yes those were kind of my thoughts. I have thought about making an FXSubject that would handle serialization and observing on the FX thread. I havent entirely dismissed that idea but in a great majority of cases, emissions going in and out should already be on the FX thread anyway. I can see occasional situations it might be useful though...

**Comment by thomasnield** [comment]
I have found a potentially serious issue: subscribing a subject does not return a `Disposable`.

**Comment by rguillens** [comment]
Explain this issue a little more. The only subscribe implementation that doesn't return a `Disposable` in `Observable` is `public final void subscribe(io.reactivex.Observer<? super T> observer)` and all `Subjects` inherit this method.

**Comment by thomasnield** [comment]
I'll post a better example tomorrow, but...

```java
Subject subject = ...
myEventsObservable.subscribe(subject); // Does not return Disposable
```

**Comment by rguillens** [comment]
I understand what you mean, a workaround could be subscribing with method references:

```
Subject subject = ...
Disposable disposable = myEventsObservable.subscribe(subject::onNext, subject::onError, subject::onComplete);
```

**Comment by thomasnield** [comment]
Yup, good thought. I looked at my implementation of CompositeObservable and that is exactly what I did.

https://github.com/ReactiveX/RxJavaFX/blob/2.x/src/main/java/io/reactivex/rxjavafx/sources/CompositeObservable.java

I'm a tad unsure what to do here... whether to tell users to do that subscribe() trick themselves, implement a new utility, or keep things as is. 

Too bad Java doesn't have extension functions like Kotlin. Those would be pretty handy for this problem right now...
